experimental/devices/cap1188/cap1188.go

```diff
@@ -24,121 +23,87 @@ import (
+	}
+	return touchStatusName[touchStatusIndex[i]:touchStatusIndex[i+1]]
+}
 
 // Dev is a handle to a cap1188.
 type Dev struct {
-	Opts
-	d             conn.Conn
-	regWrapper    mmr.Dev8
-	isSPI         bool
-	inputStatuses []TouchStatus
-	resetAt       time.Time
+	c     mmr.Dev8
+	opts  Opts
+	isSPI bool
+
+	inputStatuses [8]TouchStatus
+	lastReset     time.Time
 }
 
 func (d *Dev) String() string {
-	return fmt.Sprintf("cap1188{%s}", d.regWrapper.Conn)
+	return fmt.Sprintf("cap1188{%s}", d.c.Conn)
 }
 
 // Halt is a noop for the cap1188.
 func (d *Dev) Halt() error {
 	return nil
 }
 
-// InputStatus reads and returns the status of the 8 inputs as an array where
-// each entry indicates a touch event or not.
-func (d *Dev) InputStatus() ([]TouchStatus, error) {
-	// first check that we are ready
-	now := time.Now()
-	readyAt := d.resetAt.Add(200 * time.Millisecond)
-	if now.Before(readyAt) {
-		time.Sleep(readyAt.Sub(now))
-	}
-	// read inputs
-	status, err := d.regWrapper.ReadUint8(0x3)
+// InputStatus reads and returns the status of the 8 inputs.
+func (d *Dev) InputStatus() ([8]TouchStatus, error) {
+	d.resetSinceAtLeast(200 * time.Millisecond)
+	// Read inputs.
+	status, err := d.c.ReadUint8(0x3)
 	if err != nil {
-		return d.inputStatuses, wrap(fmt.Errorf("failed to read the input values - %s", err))
-	}
-
-	// read deltas (in two's complement, capped at -128 to 127)
-	deltasB := [nbrOfLEDs]byte{}
-	if err = d.regWrapper.ReadStruct(0x10, &deltasB); err != nil {
-		return d.inputStatuses, wrap(fmt.Errorf("failed to read the delta values - %s", err))
-	}
-	deltas := [nbrOfLEDs]int{}
-	for i, b := range deltasB {
-		deltas[i] = int(int8(b))
-	}
-
-	// read threshold
-	thresholds := [nbrOfLEDs]byte{}
-	if err = d.regWrapper.ReadStruct(0x30, &thresholds); err != nil {
-		return d.inputStatuses, wrap(fmt.Errorf("failed to read the threshold values - %s", err))
-	}
-
-	// convert the data into a sensor state
-	var touched bool
+		return d.inputStatuses, wrapf("failed to read the input values: %v", err)
+	}
+
+	// Read deltas (in two's complement, capped at -128 to 127).
+	//deltasB := [len(d.inputStatuses)]byte{}
+	//if err = d.c.ReadStruct(0x10, &deltasB); err != nil {
+	//	return d.inputStatuses, wrapf("failed to read the delta values: %v", err)
+	//}
+	//deltas := [len(d.inputStatuses)]int{}
+	//for i, b := range deltasB {
+	//	deltas[i] = int(int8(b))
+	//}
+	// Read thresholds.
+	//thresholds := [len(d.inputStatuses)]byte{}
+	//if err = d.c.ReadStruct(0x30, &thresholds); err != nil {
+	//	return d.inputStatuses, wrapf("failed to read the threshold values: %v", err)
+	//}
+
+	// Convert the data into a sensor state.
 	for i := uint8(0); i < uint8(len(d.inputStatuses)); i++ {
-		// check if the bit is set.
-		touched = (status>>(7-i))&1 == 1
-
+		// Check if the bit is set.
 		// TODO(mattetti): check if the event is passed the threshold:
 		// deltas[i] > int(thresholds[i])
-
-		if touched {
+		if (status>>(7-i))&1 != 0 {
```

Comment:
I'd suggest to at least add a comment to explain that it means that the sensored was touched

Reply:
I switched the order, as the code was doing the indexing in reverse order.

Reply:
I'm not sure what you mean but you can physically compare the output to the touched sensored/LED. I can test again the changes if you'd like